SpatialGDK/Source/SpatialGDKServices/Private/SSpatialOutputLog.cpp

```diff
@@ -238,24 +155,172 @@ void SSpatialOutputLog::PollLogFile(const FString& LogFilePath)
 	});
 }
 
-void SSpatialOutputLog::StartPollTimer(const FString& LogFilePath)
+void FSpatialLogFileReader::StartPollTimer(const FString& LogFilePath)
 {
 	// Start a timer to read the log file every PollTimeInterval seconds
 	// Timers must be started on the game thread.
 	AsyncTask(ENamedThreads::GameThread, [this, LogFilePath]
+    {
+        // It's possible that GEditor won't exist when shutting down.
+        if (GEditor != nullptr)
+        {
+            GEditor->GetTimerManager()->SetTimer(PollTimer, [this, LogFilePath]()
+            {
+                PollLogFile(LogFilePath);
+            }, PollTimeInterval, false);
+        }
+    });
+}
+
+BEGIN_SLATE_FUNCTION_BUILD_OPTIMIZATION
+void SSpatialOutputLog::Construct(const FArguments& InArgs)
+{
+	SOutputLog::Construct(SOutputLog::FArguments());
+
+	// Remove ourselves as the constructor of our parent (SOutputLog) added 'this' as a remote output device.
+	GLog->RemoveOutputDevice(this);
+
+	LogFileReaders.Add(MakeUnique<FSpatialLogFileReader>(LaunchLogFilename, [this](const FString& Content)
+	{
+		ParseLaunchLogContent(Content);
+	}));
+	LogFileReaders.Add(MakeUnique<FSpatialLogFileReader>(RuntimeLogFilename, [this](const FString& Content)
 	{
-		// It's possible that GEditor won't exist when shutting down.
-		if (GEditor != nullptr)
+		ParseSingleNodeRuntimeLogContent(Content);
```

Comment:
Since this only handles the new runtime we should bind this to the runtime switch in Nic's PR

Reply:
Yeah, probably, once we decide which PR goes in first. At the moment it doesn't matter though (this also your other comment: https://github.com/spatialos/UnrealGDK/pull/2145#discussion_r428646600).
When running on Kraken, the reg ex won't match anything and the output log looks exactly as before.